File: backend/NotesApp.Api/Models/Entities/Note.cs
namespace NotesApp.Api.Models.Entities;

public class Note
{
    public Guid Id { get; set; }

    public Guid UserId { get; set; }

    public string Title { get; set; } = string.Empty;

    public Guid? TagId { get; set; }
    public string? Content { get; set; }
    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public bool IsDeleted { get; set; }
}

File: frontend/NotesApp.Web/src/types/note.ts
export interface Note {
  id: string
  title: string
  content: string
  tagId: string | null
  tagName: string | null
  createdAt: string
  updatedAt: string | null
}

export interface CreateNoteRequest {
  title: string
  content: string
  tagId?: string | null
}

export interface UpdateNoteRequest {
  title: string
  content: string
  tagId?: string | null
}

export interface NoteQuery {
  keyword?: string
  tagId?: string
  sortBy?: string
  descending?: boolean
}
